tokio-buf/src/util/iter.rs

```diff
@@ -0,0 +1,36 @@
+use BufStream;
+use bytes::Buf;
+use futures::Poll;
+use std::marker::PhantomData;
+
+/// TODO: Dox
+pub fn iter<I, E>(i: I) -> Iter<I::IntoIter, E>
+where
+    I: IntoIterator,
+    I::Item: Buf,
+{
+    Iter {
+        iter: i.into_iter(),
+        _p: PhantomData,
+    }
+}
+
+/// TODO: Dox
+#[derive(Debug)]
+pub struct Iter<I, E> {
+    iter: I,
+    _p: PhantomData<E>,
+}
+
+impl<I, E> BufStream for Iter<I, E>
+where
+    I: Iterator,
+    I::Item: Buf,
+{
+    type Item = I::Item;
+    type Error = E;
```

Comment:
not totally sure but does it make sense to have an error here or just use a Never error type?

Reply:
Good question... having an error would mean it doesn't have to be mapped... but you are probably right.